app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,53 @@
+class ArticlesController < ApplicationController
+  
+  http_basic_authenticate_with name: "user", password: "secret", except: [:index, :show]
+
+  def index
+    @articles = Article.all
+  end
+
+  def show
+    @article = Article.find(params[:id])
+  end
+
+  def new
+    @article = Article.new
+  end
+
+  def create
+    # @article = Article.new(title: "Amazing Animals", body: "Show of top 10 Animals"title: "Amazing Animals", body: "Show of top 10 Animals")
+    @article = Article.new(article_params)
+    if @article.save
+      redirect_to(article_path(@article))
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @article = Article.find(params[:id])
+  end
+
+  def update
+    @article = Article.find(params[:id])
+
+    if @article.update(article_params)
+      redirect_to @article
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @article = Article.find(params[:id])
+    @article.destroy
```

Comment:
What happens if you try to delete an `Article` that has several `Comments` associated?

Reply:
How can you tell it will delete the `Comments` from the database?

Reply:
In the model, the parameter `dependent: :destroy` will make a DROP CASCADE query that destroys foreign keys, also the @article object selected and deleted contains the comments instances so will delete all instances.